clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/JobProvider.groovy

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.model
+
+public interface JobProvider<T extends Job> {
+  String getPlatform()
+
+  T getJob(String account, String location, String id)
```

Comment:
It's missing from Instance and Server group, and will always be known by whoever is doing the lookup. So I left it out here.